akka-http/src/main/scala/akka/http/javadsl/server/directives/SecurityDirectives.scala

```diff
@@ -259,6 +259,10 @@ abstract class SecurityDirectives extends SchemeDirectives {
   /**
    * Creates a `Basic` [[HttpChallenge]] for the given realm.
    */
-  def challengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
+  def basicChallengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
```

Comment:
If we didn't supply the method, we had the following error?
```
[info] - should define Java directive [oAuth2challengeFor] for ScalaDSL too *** FAILED *** (2 milliseconds)
[info]   akka.http.javadsl.DirectivesConsistencySpec$$anon$1: Method [akka.http.javadsl.server.directives.SecurityDirectives#oAuth2challengeFor] was not defined on class: akka.http.scaladsl.server.Directives
```
I think we need to define a Java directive as the error indicates. WDYT, @ktoso ?